src/main/java/io/github/thebusybiscuit/slimefun4/api/items/settings/FloatRangeSetting.java

```diff
@@ -0,0 +1,63 @@
+package io.github.thebusybiscuit.slimefun4.api.items.settings;
+
+import io.github.thebusybiscuit.slimefun4.api.items.ItemSetting;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.SlimefunItem;
+import org.apache.commons.lang.Validate;
+
+import javax.annotation.Nonnull;
+import javax.annotation.ParametersAreNonnullByDefault;
+
+/**
+ * This variation of {@link ItemSetting} allows you to define an {@link Float} range
+ * and enforces this range using the {@link #validateInput(Float)} method.
+ * 
+ * @author TheBusyBiscuit
+ *
+ * @see ItemSetting
+ * @see IntRangeSetting
+ * @see DoubleRangeSetting
+ */
+public class FloatRangeSetting extends ItemSetting<Float> {
```

Comment:
I do not think that Floats are properly supported here as the Config tends to auto-interpret them as doubles most of the time.
Either way, this is a potential room for error. So I suggested you to simply use an integer here.

Reply:
Sure, that is fine as well